packages/actor-query-operation-distinct-hash/lib/ActorQueryOperationDistinctHash.ts

```diff
@@ -55,8 +75,25 @@ export class ActorQueryOperationDistinctHash extends ActorQueryOperationTypedMed
       return !(hash in hashes) && (hashes[hash] = true);
     };
   }
+
+  /**
+   * Create a new distinct filter function to hash Quads.
+   * This will maintain an internal hash datastructure so that every bindings object only returns true once.
+   * @param context The action context.
+   * @return {(quad: quad) => boolean} A distinct filter for bindings.
+   */
+  public async newHashFilterQuads(context: IActionContext): Promise<(quad: Quad) => boolean> {
+    const { hashFunction } = await this.mediatorHashQuads.mediate({ allowHashCollisions: true, context });
+    const hashes: Record<string, boolean> = {};
+    return (quad: Quad) => {
+      const hash: string = hashFunction(quad);
+
+      return !(hash in hashes) && (hashes[hash] = true);
+    };
+  }
 }
 
 export interface IActorQueryOperationDistinctHashArgs extends IActorQueryOperationTypedMediatedArgs {
   mediatorHashBindings: MediatorHashBindings;
+  mediatorHashQuads: MediatorHashQuads;
```

Comment:
By making this field required, we introduce a breaking change.
We could target the `next/major` branch, but I think we can make it an optional field instead to already introduce it before the next major update.
Let's also add a `TODO` comment stating that this field must be made mandatory in the next major update.

(We can add a check on line 38 whether or not the mediator is defined)

Reply:
Done! Also had to add a check in `newHashFilterQuads`.